buildSrc/src/main/kotlin/org/gradle/plugins/compile/AvailableJavaInstallations.kt

```diff
@@ -0,0 +1,62 @@
+package org.gradle.plugins.compile
+
+import org.gradle.api.JavaVersion
+import org.gradle.internal.jvm.Jvm
+import org.gradle.jvm.toolchain.internal.JavaInstallationProbe
+import org.gradle.jvm.toolchain.internal.LocalJavaInstallation
+import java.io.File
+
+class DefaultJavaInstallation(val current: Boolean) : LocalJavaInstallation {
+    private lateinit var name: String
+    private lateinit var javaVersion: JavaVersion
+    private lateinit var javaHome: File
+    private lateinit var displayName: String
+
+    override fun getName() = name
+    override fun getDisplayName() = displayName
+    override fun setDisplayName(displayName: String) { this.displayName = displayName }
+    override fun getJavaVersion() = javaVersion
+    override fun setJavaVersion(javaVersion: JavaVersion) { this.javaVersion = javaVersion }
+    override fun getJavaHome() = javaHome
+    override fun setJavaHome(javaHome: File) { this.javaHome = javaHome }
+    fun getToolsJar(): File? = Jvm.forHome(javaHome).toolsJar
+
+    override fun toString(): String = "${displayName} (${javaHome.absolutePath})"
+}
+
+fun detectJavaInstallation(javaHome: String, javaInstallationProbe: JavaInstallationProbe): DefaultJavaInstallation {
```

Comment:
:o: This function can be `private`
:o: The return type can be omitted
:o: The function body could be a single expression

```kotlin
private
fun detectJavaInstallation(javaHomePath: String, javaInstallationProbe: JavaInstallationProbe) =
    DefaultJavaInstallation(false).apply {
        javaHome = File(javaHomePath)
        javaInstallationProbe.checkJdk(javaHome).configure(this)
    }
```

Reply:
Done.
About omitting the return type: Is this some kind of best practise in Kotlin? I sometimes like to see the return type, so I know what the function returns. Is it OK for private functions but not for public functions?

Reply:
Yup, removing spurious type information for conciseness is. If your intent is to narrow a type (e.g. return type), then it makes sense to spell it. Also note that on abstract members it is required and that the IDE will show you the inferred type at both declaration and usage sites.